source/client-side-encryption/tests/unified/namedKMS-rewrapManyDataKey.yml

```diff
@@ -0,0 +1,432 @@
+description: namedKMS-rewrapManyDataKey
+
+schemaVersion: "1.18"
+
+runOnRequirements:
+  - csfle: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents:
+        - commandStartedEvent
+  - clientEncryption:
+      id: &clientEncryption0 clientEncryption0
+      clientEncryptionOpts:
+        keyVaultClient: *client0
+        keyVaultNamespace: keyvault.datakeys
+        kmsProviders:
+          "aws:name1": { accessKeyId: { $$placeholder: 1 }, secretAccessKey: { $$placeholder: 1 } }
+          "azure:name1": { tenantId: { $$placeholder: 1 }, clientId: { $$placeholder: 1 }, clientSecret: { $$placeholder: 1 } }
+          "gcp:name1": { email: { $$placeholder: 1 }, privateKey: { $$placeholder: 1 } }
+          "kmip:name1": { endpoint: { $$placeholder: 1 } }
+          "local:name1": { key: { $$placeholder: 1 } }
+          # Use a different local master key for `local:name2`.
+          "local:name2": { key: "local+name2+YUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk" }
+          # Use a different AWS account to test wrapping between different AWS accounts.
```

Comment:
How does this use a different AWS account if aws:name2 uses the same placeholders as aws:name1?

Reply:
The test runner is expected to use different credentials for the `aws:name2` provider.

See the PR description for information on the different AWS account:

> Test credentials for `aws:name2` are available in AWS Secrets Manager under `drivers/client-side-encryption`. The `aws:name2` account credentials start with `AWSNAME2_`. See https://wiki.corp.mongodb.com/display/DRIVERS/Using+AWS+Secrets+Manager+to+Store+Testing+Secrets for more background on how the secrets are managed.

Reply:
In that case, could you update the CSE test readme to mention the aws:name2 credentials and the kmip:name1 KMS TLS options mentioned in the PR description?

Reply:
Updated.